src/components/Upload/Upload.jsx

```diff
@@ -10,33 +10,34 @@ import { Attachment } from '../Attachment'
 
 const handleAcceptedFileTypes = fileTypes => (typeof fileTypes === 'object' ? fileTypes.join(',') : fileTypes)
 
-const UploadButton = ({ variant, caption, acceptedFileTypes, multipleFiles, disabled, fileHandler, ...props }) => {
+const UploadButton = forwardRef(({ name, variant, caption, acceptedFileTypes, multipleFiles, disabled, fileHandler, ...props }, ref) => {
   const hiddenFileInput = useRef(null)
   const [uploadedFiles, setUploadedFiles] = useState()
 
-  const handleClick = () => hiddenFileInput.current.click()
-
   const handleChange = event => {
     setUploadedFiles(event.target.files)
-    fileHandler(event)
+    fileHandler ? fileHandler(event) : false
   }
 
   return (
     <Wrapper {...props}>
-      <HiddenInput
-        type='file'
-        ref={hiddenFileInput}
-        onChange={handleChange}
-        multiple={multipleFiles}
-        accept={handleAcceptedFileTypes(acceptedFileTypes)}
-      />
       <Button
-        onClick={handleClick}
         disabled={disabled}
         variant={variant === 'button' || variant === 'button-primary' ? 'filled' : 'outlined'}
         icon='upload'
         caption={caption}
-      />
+        as={!disabled && 'label'}
+        style={{display: 'flex', alignItems: 'center', justifyContent: 'center'}}
```

Comment:
Se eu não me engano, o button tem flexboxes do styeld system, dá pra passar tudo como props. Te digo isso por o padrão de código da nave evita usar style-in-line.